packages/grafana-runtime/src/types/analytics.ts

```diff
@@ -20,7 +20,8 @@ export interface DashboardInfo {
  */
 export interface DataRequestInfo extends Partial<DashboardInfo> {
   datasourceName: string;
-  datasourceId?: number;
+  datasourceId: number;
```

Comment:
Was the change from nullable intentional? :)

Reply:
Yes, when I updated the `DataSourceApi` class, I saw that the `id` field can't be null so it's more logical to have the same type here.